internal/integrationtest/util.go

```diff
@@ -108,7 +109,10 @@ func (s TransportSpec) NewServer(t *testing.T, addr string) (*yarpc.Dispatcher,
 // exercise a transport dropping connections if the transport is stopped before
 // a pending request can complete.
 func (s TransportSpec) TestConnectAndStopRoundRobin(t *testing.T) {
-	addr := "127.0.0.1:31172"
+	conn, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	addr := conn.Addr().String()
+	conn.Close()
 
 	client, rawClient := s.NewClient(t, []string{addr})
```

Comment:
If you are trying to force a connection refused, this has worked well for us (and avoids any port reuse races):
```
const _unconnectableHostPort = "0.0.0.1:1"
```

It seems like the `0.0.0.08` is reserved and it doesn't seem like listening on `0.0.0.1` allows a socket to be connected.

Reply:
Good tip. Thanks.

Reply:
unconnectableAddr works for this case. Found another fixed port later and used the listen/close trick there, because that test is for connection establishment backoff. The test simulates receiving an address from a name resolver before the host comes online.